src/test/java/services/DatafeedEventsServiceTest.java

```diff
@@ -0,0 +1,109 @@
+package services;
+
+import clients.SymBotClient;
+import clients.symphony.api.datafeeds.DatafeedClient;
+import configuration.SymConfig;
+import exceptions.APIClientErrorException;
+import model.datafeed.DatafeedV2;
+import org.junit.Before;
+import org.junit.Test;
+import org.powermock.api.mockito.PowerMockito;
+import java.util.Collections;
+import java.util.concurrent.CountDownLatch;
+import static org.mockito.Mockito.*;
+import static org.powermock.api.mockito.PowerMockito.verifyPrivate;
+
+public class DatafeedEventsServiceTest {
+
+    private SymBotClient symBotClient;
+    private DatafeedEventsService datafeedEventsService;
+    private DatafeedEventsServiceV2 datafeedEventsServiceV2;
+    private DatafeedClient datafeedClient;
+
+    @Before
+    public void initClient() {
+        symBotClient = mock(SymBotClient.class);
+        SymConfig config = mock (SymConfig.class);
+        when(config.getDatafeedVersion()).thenReturn("v2");
+        when(symBotClient.getConfig()).thenReturn(config);
+
+        datafeedClient = mock(DatafeedClient.class);
+        when(symBotClient.getDatafeedClient()).thenReturn(datafeedClient);
+        when(datafeedClient.getAckId()).thenReturn("ack_id_string");
+
+    }
+
+    @Test
+    public void readDatafeedTest() {
```

Comment:
Vertical spacing can help to read tests, for instance by adding a bank line between the Given / When / Then, i.e the setup phase, what is tested and the verifications